src/main/kotlin/carracing/game/domain/data/Car.kt

```diff
@@ -2,13 +2,45 @@ package carracing.game.domain.data
 
 private const val INITIAL_POSITION = 0
 private const val MIN_NUMBER_TO_MOVE = 4
+const val NAME_MAX_LENGTH = 5
 
-data class Car(
-    var position: Int = INITIAL_POSITION,
+class Car(
+    position: Int = INITIAL_POSITION,
+    name: String,
+    private val generateMoveNumber: () -> Int,
 ) {
-    fun move(number: Int) {
-        if (number >= MIN_NUMBER_TO_MOVE) {
+    var position: Int = position
+        private set
+
+    val name: String = validateName(name)
+
+    fun move() {
+        if (generateMoveNumber() >= MIN_NUMBER_TO_MOVE) {
             position++
         }
     }
+
+    override fun toString(): String = "Car(name='$name', position=$position)"
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is Car) return false
+
+        if (position != other.position) return false
+        return position == other.position && name == other.name
+    }
+
+    override fun hashCode(): Int {
+        var result = position
+        result = 31 * result + name.hashCode()
+        return result
+    }
```

Comment:
It's common practice to place `equals()`, `hashCode()`, and `toString()` at the end of the class.